apps/authentication-portal/src/main/webapp/basicauth.jsp

```diff
@@ -391,23 +396,46 @@
     </div>
     <% } %>
     <%!
-        private String getRecoverAccountUrl (
-            String identityMgtEndpointContext,
-            String urlEncodedURL,
-            boolean isUsernameRecovery,
-            String urlParameters) {
-
-            return identityMgtEndpointContext + "/recoveraccountrouter.do?" + urlParameters +
-                "&isUsernameRecovery=" + isUsernameRecovery + "&callback=" + Encode.forHtmlAttribute(urlEncodedURL);
+        private String getRecoverAccountUrl(String identityMgtEndpointContext, String urlEncodedURL,
+                boolean isUsernameRecovery, String urlParameters) {
+
+            return buildUrlContext(identityMgtEndpointContext) + ACCOUNT_RECOVERY_ENDPOINT_RECOVER + "?" + urlParameters
+                    + "&isUsernameRecovery=" + isUsernameRecovery + "&callback=" + Encode
+                    .forHtmlAttribute(urlEncodedURL);
+        }
+
+        private String getRegistrationUrl(String identityMgtEndpointContext, String urlEncodedURL,
+                String urlParameters) {
+
+            return buildUrlContext(identityMgtEndpointContext) + ACCOUNT_RECOVERY_ENDPOINT_REGISTER + "?"
+                    + urlParameters + "&callback=" + Encode.forHtmlAttribute(urlEncodedURL);
         }
 
-        private String getRegistrationUrl (
-            String identityMgtEndpointContext,
-            String urlEncodedURL,
-            String urlParameters) {
+        private String buildUrlContext(String identityMgtEndpointContext) {
 
-            return identityMgtEndpointContext + "/register.do?" + urlParameters +
-                "&callback=" + Encode.forHtmlAttribute(urlEncodedURL);
+            String url;
+            if (StringUtils.isNotBlank(identityMgtEndpointContext)) {
+                url = identityMgtEndpointContext;
+            } else {
+                url = buildTenantQualifiedUrl(ACCOUNT_RECOVERY_ENDPOINT);
```

Comment:
This would still need the additional query parameters appended other than tenant